usecases/mu/mpc/testhelper_test.go

```diff
@@ -0,0 +1,118 @@
+
+	s.service = service.NewService(configuration, serviceHandler)
+	_ = s.service.Setup()
+
+	mockRemoteDevice := spinemocks.NewDeviceRemoteInterface(s.T())
+	s.mockRemoteEntity = spinemocks.NewEntityRemoteInterface(s.T())
+	mockRemoteFeature := spinemocks.NewFeatureRemoteInterface(s.T())
+	mockRemoteDevice.EXPECT().FeatureByEntityTypeAndRole(mock.Anything, mock.Anything, mock.Anything).Return(mockRemoteFeature).Maybe()
+	mockRemoteDevice.EXPECT().Ski().Return(remoteSki).Maybe()
+	s.mockRemoteEntity.EXPECT().Device().Return(mockRemoteDevice).Maybe()
+	s.mockRemoteEntity.EXPECT().EntityType().Return(mock.Anything).Maybe()
+	entityAddress := &model.EntityAddressType{}
+	s.mockRemoteEntity.EXPECT().Address().Return(entityAddress).Maybe()
+	mockRemoteFeature.EXPECT().DataCopy(mock.Anything).Return(mock.Anything).Maybe()
+	mockRemoteFeature.EXPECT().Address().Return(&model.FeatureAddressType{}).Maybe()
+	mockRemoteFeature.EXPECT().Operations().Return(nil).Maybe()
+
+	localEntity := s.service.LocalDevice().EntityForType(model.EntityTypeTypeInverter)
+	s.sut, _ = NewMPC(
+		localEntity,
+		s.Event,
+		&MonitorPowerConfig{
+			ConnectedPhases:   ConnectedPhasesABC,
+			ValueSourceTotal:  util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+			ValueSourcePhaseA: util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+			ValueSourcePhaseB: util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+			ValueSourcePhaseC: util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+		},
+		&MonitorEnergyConfig{
+			ValueSourceProduction:  util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+			ValueSourceConsumption: util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+		},
+		&MonitorCurrentConfig{
+			ValueSourcePhaseA: util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+			ValueSourcePhaseB: util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+			ValueSourcePhaseC: util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+		},
+		&MonitorVoltageConfig{
+			SupportPhaseToPhase:  true,
+			ValueSourcePhaseA:    util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+			ValueSourcePhaseB:    util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+			ValueSourcePhaseC:    util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+			ValueSourcePhaseAToB: util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+			ValueSourcePhaseBToC: util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+			ValueSourcePhaseCToA: util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+		},
+		&MonitorFrequencyConfig{
+			ValueSource: util.Ptr(model.MeasurementValueSourceTypeMeasuredValue),
+			ValueConstraints: util.Ptr(model.MeasurementConstraintsDataType{
+				ValueRangeMin: model.NewScaledNumberType(0),
+				ValueRangeMax: model.NewScaledNumberType(100),
+				ValueStepSize: model.NewScaledNumberType(1),
+			}),
+		},
+	)
+	s.sut.AddFeatures()
+	s.sut.AddUseCase()
+
+	//s.remoteDevice, s.monitoredEntity = setupDevices(s.service, s.T())
```

Comment:
I would love to see such tests as in CS/LPC to be added, not just using the mocks.

Reply:
We updated the tests to use Filters to query the stored data representation for the usecases. This should test that querying the data works from the client perspective without needing to reimplement the ma/mpc use cases in the tests for mu/mpc.